lib/form_validators.dart

```diff
@@ -1,24 +1,30 @@
 import 'package:easy_localization/easy_localization.dart';
 import 'package:water_tracker/generated/locale_keys.g.dart';
 
-typedef ValidatorFunction = String? Function(String?);
+typedef SingleValidatorFunction = String? Function(String?);
+typedef DoubleValidatorFunction = String? Function(String?, String?);
 
 class FormValidators {
   static final _emailRegexp = RegExp(r'^[\w-\.]+@([\w-]+\.)+[\w]{2,4}');
-  static final _passwordRegexp = RegExp(r'^[\w-\.]+@([\w-]+\.)+[\w]{2,4}');
+  static final _passwordRegexp = RegExp(r'^(?=.*?[A-Z])(?=.*?[a-z]).{8,}$');
 
-  static ValidatorFunction emailValidator = (value) {
+  static SingleValidatorFunction emailValidator = (value) {
     if (value == null || value.isEmpty) return LocaleKeys.enter_your_email.tr();
     if (!_emailRegexp.hasMatch(value)) return LocaleKeys.please_enter_correct_email.tr();
     return null;
   };
 
-  static ValidatorFunction passwordValidator = (value) {
+  static SingleValidatorFunction passwordValidator = (value) {
     if (value == null || value.isEmpty) return LocaleKeys.please_enter_password.tr();
     if (!_passwordRegexp.hasMatch(value)) {
-      return LocaleKeys.invalid_password;
+      return LocaleKeys.invalid_password.tr();
```

Comment:
Because this is a validation for the correct password, if the user tries to log in, and not a registration validation, I will leave todo to make a separate function for this